tasks/main.yml

```diff
@@ -106,6 +107,38 @@
   tags:
     - wg-config
 
+- name: Create private key for unmanaged hosts
+  shell: "wg genkey | tee {{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+  args:
+    creates: "{{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+  register: uh_privkey
+  with_items: "{{ wireguard_unmanaged_hosts | default([]) }}"
+
+- name: Validate permissions of unmanaged hosts' private keys
+  file:
+    path: "{{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+    mode: '0400'
+  with_items: "{{ wireguard_unmanaged_hosts | default([]) }}"
+
+- name: Recover existing private key for unmanaged hosts
+  shell: "cat {{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
```

Comment:
my only concern here is uh_pubkey can be fill either by key generation, either by key recovery.
if using slurp, output format will change and more processing of data after.
advice?

Reply:
TBH I'm not sure if I understand your question. This task just assigns a variable value. The needed base64 encoding/decoding is not worth talking about IMHO. If you would run this task 10x / sec. it would be different of course. Forking a shell also costs resources maybe even more then the Jinja "b64decode" filter.

Reply:
uh_private_key is output of two tasks:
* Create private key for unmanaged host: change if priv key file does not exist
* Recover existing private key for unmanaged hosts. change_when: false
for which format is the same.

if moving to slurp, we get key only from latter. is that what is wanted?